File: 400/413 - Arithmetic Slices.c
int numberOfArithmeticSlices(int* A, int ASize) {
    int d, dp, cnt = 2, ret = 0, i;
    
    if (ASize < 3)
        return 0;
    dp = A[1] - A[0];
    
    for (i = 2; i < ASize; ++i) {
        d = A[i] - A[i - 1];
        
        if (d == dp)
            ++cnt;
        else {
            if (cnt > 2)
                ret += (cnt - 1) * (cnt - 2) / 2;
            cnt = 2;
        }
        
        dp = d;
    }
    if (cnt > 2)
        ret += (cnt - 1) * (cnt - 2) / 2;
    
    return ret;
}

File: csharp/413.arithmetic-slices.cs
/*
 * @lc app=leetcode id=413 lang=csharp
 *
 * [413] Arithmetic Slices
 */
public class Solution {
    public int NumberOfArithmeticSlices(int[] A) {
        int len = 0, ret = 0;
        for (int i = 2; i < A.Length; ++i) {
            if ((long) A[i] - (long) A[i - 1] == (long) A[i - 1] - (long) A[i - 2]) {
                ret += ++len;
            } else {
                len = 0;
            }
        }
        return ret;
    }
}

